Enable "swift test" command

AnthonyOliveri opened the pull request:
Fixes https://github.com/evgenyneu/keychain-swift/issues/112

When I forked this project, I ran into the issue linked above when running `swift test` on my Mac. This PR fixes the tests by adding a test target to Package.swift, and adding import statements to all of the test files.

One thing to note is that `testClear()` fails with error code ` -25244` (errSecInvalidOwnerEdit). I believe this is due to the fact that `clear()` is attempting to delete everything stored in the Mac keychain (which would obviously be a bad thing). Fortunately, keychain permissions prevent this from happening with the aforementioned error. Not sure how to get that test to pass.

evgenyneu commented:
Thanks for the update. I noticed a problem, unit tests for the `macOS Tests` target no longer run in the Xcode, showing error `No such module 'KeychainSwift'` for the line `@testable import KeychainSwift`.

> One thing to note is that testClear() fails with error code -25244 (errSecInvalidOwnerEdit). I believe this is due to the fact that clear() is attempting to delete everything stored in the Mac keychain (which would obviously be a bad thing).

You are right. Currently, there is a separate target `macOS Tests` for running tests on Mac in Xcode in order to solve this issue. Files `ClearTests.swift` and `ConcurrencyTests.swift` are not included in this target. I wonder if there is a way to run tests in similar way using Swift Package Manager: run tests in `KeychainSwiftTests` on iOS, but on Mac run tests  from `macOS Tests` target instead?

AnthonyOliveri commented:
Good catch on the Xcode test failures. I fixed those (verifying that all targets build and run tests successfully) and added the macOS tests to the KeychainSwift-macOS target so the tests can be run from there as well.

I also excluded the "ClearTests.swift" file from the SPM test target, not just for macOS. I figured that it was safer to not run those tests on any platform that a user may call `swift test` from, since it could potentially delete important items if keychain access permissions aren't set up properly.

evgenyneu commented:
Thanks for your help, very good stuff! Released your changes in version `17.0.0`.

AnthonyOliveri commented:
Great! I'll have another PR soon for adding a `getAll()` method.

evgenyneu commented:
> Great! I'll have another PR soon for adding a getAll() method.

Thanks, could explain this update before investing your time in it please? :)

AnthonyOliveri commented:
Sure thing. Should I open an issue for it?

evgenyneu commented:
Yeah, good idea, thanks.

AnthonyOliveri commented:
https://github.com/evgenyneu/keychain-swift/issues/116
